packages/react/src/useFetch.ts

```diff
@@ -112,7 +124,8 @@ export function useFetch(path: string, options?: FetchHookOptions): FetchHookRes
 export function useFetch<T = string>(path: string, options?: FetchHookOptions): FetchHookResult<T> {
   // Used to prevent state update if the component is unmounted
   const mounted = useRef<boolean>(true);
-  const memoizedOptions = useStructuralMemo<FetchHookOptions>(options ?? {});
+  const { onStreamComplete, ...optionsToMemoize } = options ?? {};
```

Comment:
need to not pass this into useStructuralMemo, since otherwise will lead to a new value on every render. Is there a pattern to make sure this function is memoized and if not memoize it?

Reply:
So the purpose of `useStructuralMemo` is to _not_ produce a new identity if the new value is structurally the same as the previous. This should avoid re-renders.

I don't think the issue is actually `useStructuralMemo`, which I recommend you keep, but with the caller passing in a new instance of `onStreamComplete` every time. A couple of thoughts:

1. This is an issue in most libraries, where you're expected to `useCallback` when passing things into library components.
2. Find out if there's a way we can safely compare two functions (doubtful it's perfect, but we could compare their `toString()`s.

---

And just a little side note on why we introduced `useStructuralMemo`: to make things avoid re-rendering in the usual case of how users will likely write this code:

```js
// Users are likely to write this
useFetch("...", { ...options })

// But to avoid re-renders without (structural memo) they'd have to do something like this
const options = useMemo(() => { ...options}, [...deps])
useFetch("...", options)
```

Reply:
Oh, I completely missed that right below this comment was a `useStructuralMemo` 🤦